examples/map/multiproc_map/example.py

```diff
@@ -11,23 +12,32 @@ def is_prime(n):
         return True
 
 
-def my_handler(keys: list[str], datum: Datum) -> Messages:
-    val = datum.value
-    _ = datum.event_time
-    _ = datum.watermark
-    messages = Messages()
-    for i in range(2, 100000):
-        is_prime(i)
-    messages.append(Message(val, keys=keys))
-    return messages
+class PrimeMap(Mapper):
+    """
+    This class needs to be of type Mapper class to be used
+    as a handler for the MapServer class.
+    Example of a mapper that calculates if a number is prime.
+    """
+
+    def handler(self, keys: list[str], datum: Datum) -> Messages:
+        val = datum.value
+        _ = datum.event_time
+        _ = datum.watermark
+        messages = Messages()
+        for i in range(2, 100000):
+            is_prime(i)
+        messages.append(Message(val, keys=keys))
+        return messages
 
 
 if __name__ == "__main__":
     """
     Example of starting a multiprocessing map vertex.
-    To enable set the env variable
-        MAP_MULTIPROC="true"
-    in the pipeline config for the numa container.
     """
-    grpc_server = MultiProcMapper(handler=my_handler)
+    # To set the env server_count value set the env variable
+    # NUM_CPU_MULTIPROC="N"
+    server_count = int(os.getenv("NUM_CPU_MULTIPROC", "2"))
+    prime_class = PrimeMap()
+    # Server count is the number of server processes to start
+    grpc_server = MapMultiprocServer(prime_class, server_count=server_count)
```

Comment:
```suggestion
    # Server count is the number of server processes to start
# non-thread safe
grpc_server = MapMultiprocServer(instance=PrimeMap(), server_count=server_count)
# thread safe since we create an instance for every message
grpc_server = MapMultiprocServer(class=PrimeMap, server_count=server_count)
```

don't implement but just think